File: java2csharp/java/Samsung/ProfessionalAudio.FloatingConotrller.WidgetsLibrary/refactor/LogUtils.java
package com.samsung.android.sdk.professionalaudio.widgets.refactor;

import android.util.Log;

/**
 * @brief Log throwable objects
 */
public class LogUtils {

    /**
     * Log throwable object.
     *
     * @param tag
     * @param msg
     * @param ex
     */
    public static void throwable(String tag, String msg, Throwable ex) {
        StringBuilder stackTrace = new StringBuilder();
        stackTrace.append(msg)
                .append(": ")
                .append(ex.toString());

        if (null != ex.getMessage()) {
            stackTrace.append(" (")
                    .append(ex.getMessage())
                    .append(")");
        }

        stackTrace.append("\n");
        for (StackTraceElement ste : ex.getStackTrace()) {
            stackTrace.append("\tat ")
                    .append(ste.getClassName())
                    .append(".")
                    .append(ste.getMethodName())
                    .append(" (")
                    .append(ste.getFileName())
                    .append(":")
                    .append(ste.getLineNumber())
                    .append(")\n");
        }
        Log.e(tag, stackTrace.toString());
    }

}

File: java2csharp/csharp/Samsung/ProfessionalAudio.FloatingConotrller.WidgetsLibrary/refactor/LogUtils.cs
using System;
using System.Text;

namespace com.samsung.android.sdk.professionalaudio.widgets.refactor
{

	using Log = android.util.Log;

	/// <summary>
	/// @brief Log throwable objects
	/// </summary>
	public class LogUtils
	{

		/// <summary>
		/// Log throwable object.
		/// </summary>
		/// <param name="tag"> </param>
		/// <param name="msg"> </param>
		/// <param name="ex"> </param>
		public static void throwable(string tag, string msg, Exception ex)
		{
			StringBuilder stackTrace = new StringBuilder();
			stackTrace.Append(msg).Append(": ").Append(ex.ToString());

			if (null != ex.Message)
			{
				stackTrace.Append(" (").Append(ex.Message).Append(")");
			}

			stackTrace.Append("\n");
			foreach (StackTraceElement ste in ex.StackTrace)
			{
				stackTrace.Append("\tat ").Append(ste.ClassName).Append(".").Append(ste.MethodName).Append(" (").Append(ste.FileName).Append(":").Append(ste.LineNumber).Append(")\n");
			}
			Log.e(tag, stackTrace.ToString());
		}

	}

}